include/ff_error.h

```diff
@@ -62,6 +62,7 @@
 #define FF_GETMOD_FUNC( x )    ( ( ( ( unsigned ) x ) >> FF_FUNCTION_SHIFT ) & 0xFFFF )
 #define FF_ERRFLAG    0x80000000
 #define FF_isERR( x )          ( ( ( x ) & FF_ERRFLAG ) != 0 )
+#define FF_createERR( x, id ) ( ( FF_Error_t ) ( ( x ) |  ( id ) ) )
```

Comment:
Note - added `FF_createERR` to ensure the errors are always created the same way.  This allows easier updates if at some point in the future want to ensure x  and id are a particular enumeration type.

Reply:
That's a good choice, thank you